File: services/auth-policy/src/main/kotlin/kr/co/a4ai/gcssaker/authpolicy/api/ApiRoutes.kt
package kr.co.a4ai.gcssaker.authpolicy.api

object AuthApiRoutes {
    const val ROOT = "/auth"
    const val SIGNUP = "/signup"
    const val LOGIN = "/login"
    const val REFRESH = "/refresh"
    const val ME = "/me"
    const val LOGOUT = "/logout"
}

object HealthApiRoutes {
    const val HEALTHZ = "/healthz"
    const val READYZ = "/readyz"
}

object StreamPolicyApiRoutes {
    const val ROOT = "/policy/streams"
    const val ACCESS = "/access"
}

object OpsApiRoutes {
    const val EVENTS = "/ops/events"
    const val TIME_STATUS = "/ops/time/status"
    const val TIME_CHECK = "/ops/time/check"
    const val TIME_CONFIG = "/ops/time/config"
}

object OperationalReadApiRoutes {
    const val TELEMETRY_ALL = "/telemetry/all"
    const val TELEMETRY_INGEST = "/telemetry/"
    const val ASSET_BY_GATEWAY = "/asset/{gatewayUuid}"
}

File: gcs-dashboard/src/features/apiRoutes.ts
export const AUTH_ROUTES = Object.freeze({
  login: "/login",
  refresh: "/refresh",
  logout: "/logout",
  signup: "/signup",
  me: "/me",
});

export const BACKEND_ROOT_ROUTES = Object.freeze({
  healthz: "/healthz",
  readyz: "/readyz",
  adminDevices: "/auth-policy/api/v1/devices",
  groups: "/auth-policy/api/v1/groups",
  provisioningTokens: "/auth-policy/api/v1/provisioning-tokens",
  signupTokens: "/auth-policy/api/v1/signup-tokens",
  mediaControlHealthz: "/media-control/healthz",
  mediaControlReadyz: "/media-control/readyz",
  streamStatus: "/stream/status",
});

export const DASHBOARD_API_ROUTES = Object.freeze({
  assetByGatewayPrefix: "/asset/",
  operationalEvents: "/ops/events",
  operationalEventsPage: "/ops/events/page",
  operationalEventsStream: "/ops/events/stream",
  operationalEventMetrics: "/ops/events/metrics",
  operationalEventBuckets: "/ops/events/buckets",
  serverHealthSnapshots: "/ops/server-health/snapshots",
  streamSessions: "/ops/stream-sessions",
  streamSessionsStream: "/ops/stream-sessions/stream",
  telemetryAll: "/telemetry/all",
  telemetryIngest: "/telemetry/",
  telemetryHistorySuffix: "/history",
  timeSyncStatus: "/ops/time/status",
  timeSyncCheck: "/ops/time/check",
  timeSyncConfig: "/ops/time/config",
});

export const STREAM_API_ROUTES = Object.freeze({
  streams: "/streams",
  iceServers: "/streams/ice-servers",
  mapConfig: "/map/config",
});
